src/etc/inc/dyndns.class

```diff
@@ -882,6 +921,191 @@
+			}
+			
+			switch ($this->_dnsService) {
+				case 'dreamhost':
+				case 'dreamhost-v6':
+					$server = 'https://api.dreamhost.com/';
+					$post_data['key'] = $this->_dnsPass;
+					$post_data['unique_id'] = uniqid($this->_dnsHost);
+					$post_data['cmd'] = 'dns-remove_record';
+					$post_data['format'] = 'json';
+					$post_data['value'] = $existing_ip;
+					$post_data['record'] = $this->_dnsHost;
+					$isv6 = ($this->_dnsService === 'dreamhost-v6');
+					$post_data['type'] = $isv6 ? 'AAAA' : 'A';
+					$port = "";
+					if ($this->_dnsServer) {
+						$server = $this->_dnsServer;
+					}
+					if ($this->_dnsPort) {
+						$port = ":" . $this->_dnsPort;
+					}
+					curl_setopt($ch, CURLOPT_URL, $server . $port);
+					curl_setopt($ch, CURLOPT_POSTFIELDS, $post_data);
+					break;
+				default:
+					break;
+				}
+				if ($this->_dnsService != 'ods') {
+					curl_setopt($ch, CURLOPT_HEADER, 1);
+					$response = curl_exec($ch);
+					$header_size = curl_getinfo($ch, CURLINFO_HEADER_SIZE);
+					$header = substr($response, 0, $header_size);
+					$data = substr($response, $header_size);
+					$this->_checkStatus($ch, $data, $header);
+					@curl_close($ch);
+				}
+			}
+		
+		/**
+		 * Private Function (added 23 Feb 17)
+		 * Retrieves current DNS records.
+		 * 
+		 * Some services cannot perform new operations without the caller 
+		 * providing existing record information. 
+		 */
+		function _lookup_current() {
+			if ($this->_dnsVerboseLog) {
+				log_error(sprintf(gettext('Dynamic DNS %1$s (%2$s): _listCurrent() starting.'), $this->_dnsService, $this->_FQDN));
+			}
+				
+			if (strstr($this->_dnsRequestIf, "_vip")) {
+				$parentif = get_configured_vip_interface($this->_dnsRequestIf);
+				$realparentif = convert_friendly_interface_to_real_interface_name($parentif);
+			} else {
+				$realparentif = $this->_dnsRequestIf;
+			}
+				
+			$ch = curl_init();
+				
+			if ($this->_useIPv6 == false) {
+				curl_setopt($ch, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
+			}
+				
+			if ($this->_dnsService != 'ods') {
+				curl_setopt($ch, CURLOPT_USERAGENT, $this->_UserAgent);
+				curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+				curl_setopt($ch, CURLOPT_INTERFACE, 'if!' . $realparentif);
+				curl_setopt($ch, CURLOPT_TIMEOUT, 120); // Completely empirical
+			}
+				
+			switch ($this->_dnsService) {
+				case 'dreamhost':
+				case 'dreamhost-v6':
+					$server = 'https://api.dreamhost.com/';
+					$post_data['key'] = $this->_dnsPass;
+					$post_data['unique_id'] = uniqid($this->_dnsHost);
+					$post_data['cmd'] = 'dns-list_records';
+					$post_data['format'] = 'json';
+					$port = "";
+					if ($this->_dnsServer) {
+						$server = $this->_dnsServer;
+					}
+					if ($this->_dnsPort) {
+						$port = ":" . $this->_dnsPort;
+					}
+					curl_setopt($ch, CURLOPT_URL, $server . $port);
+					curl_setopt($ch, CURLOPT_POSTFIELDS, $post_data);
+					break;
+				default:
+					break;
+			}
+			if ($this->_dnsService != 'ods') {
```

Comment:
Is there a way to get the "old" WAN IP address or are we way beyond that at this point?  I was wondering how else I could obtain the previous WAN address rather than having to perform a complete API call.  In the short term I have added a flag in a similar way to above.